packages/grzctl/src/grzctl/commands/list_submissions.py

```diff
@@ -3,27 +3,100 @@
+    while also handling missing submissions and submissions without a state yet.
+    """
+    submission_from_db = submission_db.get_submission(submission_id)
+    if submission_from_db:
+        latest_state = submission_from_db.get_latest_state()
+        if latest_state is None:
+            latest_state_txt = rich.text.Text("none", style="italic yellow")
+        else:
+            latest_state_str = latest_state.state.value
+            latest_state_txt = (
+                rich.text.Text(latest_state_str, style="red")
+                if latest_state_str == SubmissionStateEnum.ERROR
+                else rich.text.Text(latest_state_str)
+            )
+    else:
+        latest_state_txt = rich.text.Text("missing", style="italic yellow")
+
+    return latest_state_txt
+
+
+def _prepare_table(summaries: list[InboxSubmissionSummary], config: ListConfig) -> rich.table.Table:
+    """
+    Constructs a nice Rich Table to display inbox status and database state of submissions in the inbox.
+    """
+    table = rich.table.Table()
+    table.add_column("ID", no_wrap=True)
+    table.add_column("Inbox Status", no_wrap=True, justify="center")
+    if config.db is not None:
+        table.add_column("Database State", no_wrap=True, justify="center", style="green")
+        submission_db = get_submission_db_instance(db_url=config.db.database_url)
+    table.add_column("Oldest Upload", overflow="fold")
+    table.add_column("Newest Upload", overflow="fold")
+    for summary in summaries:
+        match summary.state:
+            case InboxSubmissionState.INCOMPLETE:
+                status_text = rich.text.Text("Incomplete", style="yellow")
+            case InboxSubmissionState.COMPLETE:
+                status_text = rich.text.Text("Complete", style="green")
+            case InboxSubmissionState.CLEANING:
+                status_text = rich.text.Text("Cleaning", style="yellow")
+            case InboxSubmissionState.CLEANED:
+                status_text = rich.text.Text("Cleaned", style="sky_blue1")
+            case InboxSubmissionState.ERROR:
+                status_text = rich.text.Text("Error", style="red")
+            case _:
+                status_text = rich.text.Text("Unknown", style="red")
+        row: list[rich.console.RenderableType] = [
+            summary.submission_id,
+            status_text,
+            summary.oldest_upload.astimezone().strftime("%Y-%m-%d %H:%M:%S"),
+            summary.newest_upload.astimezone().strftime("%Y-%m-%d %H:%M:%S"),
+        ]
+        if submission_db is not None:
+            latest_state_txt = _get_latest_state_txt(submission_db, summary.submission_id)
+            row.insert(2, latest_state_txt)
+        table.add_row(*row)
+    return table
+
+
+def _validate_limit(ctx: click.Context, param: click.Parameter, value: int):
+    if value < 0:
+        raise click.BadParameter("limit must be a positive integer")
+
+    return value
```

Comment:
use https://click.palletsprojects.com/en/stable/api/#click.IntRange instead?

Reply:
Didn't know this existed, thanks!